src/features/layout/update/updateFormLayoutSagas.ts

```diff
@@ -279,13 +282,21 @@ export function* updateRepeatingGroupsSaga({
   }
 }
 
+const UIConfigSelector: (store: IRuntimeStore) => IUiConfig = (store: IRuntimeStore) => store.formLayout.uiConfig;
+
 export function* updateCurrentViewSaga({
   payload: { newView, runValidations, returnToView, skipPageCaching, keepScrollPos, focusComponentId },
 }: PayloadAction<IUpdateCurrentView>): SagaIterator {
   try {
-    // When triggering navigation to the next page, we need to make sure there are no unsaved changes. The action to
-    // save it should be triggered elsewhere, but we should wait until the state settles before navigating.
-    yield waitFor((state) => !state.formData.unsavedChanges);
+    const uiConfig: IUiConfig = yield select(UIConfigSelector);
+
+    if (uiConfig.autoSaveBehavior === 'onChangePage') {
+      yield call(putFormData, {});
```

Comment:
This does not take into account stateless apps, which run a different saving routine. Also, errors from this function will throw. Take a look at `saveFormDataSaga` and see if you can call that instead (or better yet, `yield put(FormDataActions.saveLatest({}));`)